pkg/hubble/relay/observer.go

```diff
@@ -88,14 +95,47 @@ func (s *Server) GetFlows(req *observerpb.GetFlowsRequest, stream observerpb.Obs
 			}
 		})
 	}
+
+	pq := queue.NewPriorityQueue(qlen)
+	sortedFlows := make(chan *observerpb.GetFlowsResponse, qlen)
+	go func() {
+		defer close(sortedFlows)
+	flowsLoop:
+		for {
+			select {
+			case flow, ok := <-flows:
+				if !ok {
+					break flowsLoop
+				}
+				if pq.Len() == qlen {
+					f := pq.Pop()
+					select {
+					case sortedFlows <- f:
+					case <-ctx.Done():
+						return
+					}
+				}
+				pq.Push(flow)
+			case <-time.After(1 * time.Second): // make sure to drain the queue when no new flow responses are received
+				if f := pq.Pop(); f != nil {
+					select {
+					case sortedFlows <- f:
+					case <-ctx.Done():
+						return
+					}
+				}
+			}
+		}
+		// drain the queue
+		for f := pq.Pop(); f != nil; f = pq.Pop() {
+			sortedFlows <- f
```

Comment:
I believe we have to check for `case <-ctx.Done()` as well.

Reply:
This won't work because `g.Wait()` calls the `cancel` func which would cause the routine to be terminated prematurely in a normal case (ie when `server.Stop()` is not called).

However, there is indeed an issue here but I also cannot make this routine part of the errgroup as the only way to safely close the `flows` channel, that has multiple writers, is to ensure they are all done. Mmmh, maybe I could solve this with an atomic counter and make the routine that sorts is part of the errgroup.